File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/MobileConfigurations/AccountOpening/AccountOpening.cs
namespace Psi.Data.Models.ClientConfigurationModels.MobileConfigurations.AccountOpening
{
    public class AccountOpening : SettingsBaseHelper
    {
        private readonly ISettingsBase _settingsBase;
        private JoinCreditUnion.JoinCreditUnion _joinCreditUnion;
        private MiniOao.MiniOao _miniOao;

        public AccountOpening(ISettingsBase settingsBase) : base(settingsBase)
        {
            _settingsBase = settingsBase;
        }

        public JoinCreditUnion.JoinCreditUnion JoinCreditUnion
        {
            get { return _joinCreditUnion ?? (_joinCreditUnion = new JoinCreditUnion.JoinCreditUnion(_settingsBase)); }
            set { _joinCreditUnion = value; }
        }

        public MiniOao.MiniOao MiniOao
        {
            get { return _miniOao ?? (_miniOao = new MiniOao.MiniOao(_settingsBase)); }
            set { _miniOao = value; }
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/Psi/Data/Models/ClientConfigurationModels/MobileConfigurations/AccountOpening/AccountOpening.ts
import { Setting, ISettingsGroup, ISettingsMetadata } from '@models/base/types';
import { JoinCreditUnion } from './JoinCreditUnion.JoinCreditUnion';
import { MiniOao } from './MiniOao.MiniOao';
export interface AccountOpeningConfig {
    JoinCreditUnion: JoinCreditUnion;
    MiniOao: MiniOao;
}

export class AccountOpening implements ISettingsGroup {
    static readonly metadata: ISettingsMetadata = {
        groupName: 'AccountOpening'
    };


            private _joinCreditUnion: JoinCreditUnion;
            get joinCreditUnion(): JoinCreditUnion {
                return this._joinCreditUnion;
            }
            set joinCreditUnion(value: JoinCreditUnion) {
                this._joinCreditUnion = value;
            }

            private _miniOao: MiniOao;
            get miniOao(): MiniOao {
                return this._miniOao;
            }
            set miniOao(value: MiniOao) {
                this._miniOao = value;
            }


    constructor() {}


        toSettings(): Setting[] {
            return [
                { key: "AccountOpening.JoinCreditUnion", value: this._joinCreditUnion, dataType: 'joincreditunion.joincreditunion', label: "Join Credit Union" },
                { key: "AccountOpening.MiniOao", value: this._miniOao, dataType: 'minioao.minioao', label: "Mini Oao" },
            ];
        }

}